pkg/controller/statusmanager/status_manager.go

```diff
@@ -158,15 +159,32 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...condi
 			},
 		)
 	} else if reachedAvailableLevel {
-		// If successfully deployed all components and is not failing on anything, mark as Available
-		status.eventEmitter.EmitAvailableForConfig()
-		conditionsv1.SetStatusConditionNoHeartbeat(&config.Status.Conditions,
-			conditionsv1.Condition{
-				Type:   conditionsv1.ConditionAvailable,
-				Status: corev1.ConditionTrue,
-			},
-		)
-		config.Status.ObservedVersion = operatorVersion
+		if status.isRunningOnOpenshift411OrLater() && config.Spec.NMState != nil {
+			// In case the clusterVersion.Status.Desired.Version is 4.11 and we're deploying nmstate
+			// with CNAO, we'll set degraded state to block CNV upgrade, forcing migration of
+			// nmstate to kubernetes nmstate operator
+			reason := "InvalidConfiguration"
+			message := "NMState deployment is not provided in CNV >= 4.11, please install Kubernetes NMState Operator"
+			status.eventEmitter.EmitFailingForConfig(reason, message)
+			conditionsv1.SetStatusConditionNoHeartbeat(&config.Status.Conditions,
+				conditionsv1.Condition{
+					Type:    conditionsv1.ConditionDegraded,
+					Status:  corev1.ConditionTrue,
+					Reason:  reason,
+					Message: message,
+				},
+			)
+		} else {
+			// If successfully deployed all components and is not failing on anything, mark as Available
+			status.eventEmitter.EmitAvailableForConfig()
+			conditionsv1.SetStatusConditionNoHeartbeat(&config.Status.Conditions,
```

Comment:
We should change heratbeat if previous condition was not success, although I don't know if we can check that or if it's worth it.

Reply:
imo if we don't keep track about the heartbeat anymore then there's no need to update it in any scenario - neither good or bad.
We do have the TransitionTimestamp (when the status last changed value) which might be enough..

Reply:
Ah, so we don't want to update it in https://github.com/kubevirt/cluster-network-addons-operator/pull/1292#discussion_r839518033 either, correct?

Reply:
Then feels like heartbeat is kind of obsolete and replaced by TransitionTimestamp ?

Reply:
it doesn't replace `TransitionTimestamp` becasue `TransitionTimestamp` is already there.
heartbeat is no longer published on CNAO. This is because it also causes a resourceVersion change (every 1 minute) and that triggers a repeating reconcile on HCO. As a result, we decided to remove it entirely on main and release-0.65 branches.